server/src/controller/todos.ts

```diff
@@ -0,0 +1,136 @@
+import { Request, Response } from "express";
+import { TodoService } from "../service/todo";
+import TodoModel, { Todo } from "../models/todo";
+import { get } from "lodash";
+import { buildSortPipeline } from "../helpers/index";
+
+export class TodoController {
+  static async addTodo(req: Request, res: Response) {
+    try {
+      const userId = get(req, "identity._id") as string;
+      const { title, description, isChecked, priority, date } = req.body;
+
+      if (!title || !description || !isChecked || !priority || !date) {
+        return res.sendStatus(400);
+      }
+
+      const todo = await TodoService.createTodo({
+        userId,
+        title,
+        description,
+        priority,
+        isChecked,
+        date,
+      });
+
+      return res.status(201).json(todo);
+    } catch (error) {
+      console.error("Error during registration:", error);
+      return res.sendStatus(500);
+    }
+  }
+
+  static async getTodos(req: Request, res: Response) {
+    try {
+      const userId = get(req, "identity._id") as string;
+      const query = req.query.query as string;
+      const isAscending = req.query.is_ascending === "true";
+      const filterBy = req.query.filter_by as
+        | "title"
+        | "description"
+        | "date"
+        | "priority";
+
+      if (!isAscending) {
```

Comment:
so this will throw an error if you try to sort in descending order ?

Reply:
oops